### ISSUE TITLE
Feature Request: Azure Firewall: Specify TLS Certificate Location in Azure Keyvault

### ISSUE THREAD

**Issue by archmangler** [Azure/terraform-azurerm-caf-enterprise-scale#513]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### Is your feature request related to a problem?

- For TLS encryption, enterprises would like to specify a CA certificate generated by a third party provider and stored in Azure Keyvault, which is then configured in Azure Firewall as the certificate for TLS traffic encryption.
- The current configuration scheme doesn't appear to support a mechanism for specifying the AKV vault and path to access the Certificate from upon deploy or update.:

```
          azure_firewall = object({
              enabled = bool
              config = object({
                address_prefix                = string
                enable_dns_proxy              = bool
                dns_servers                   = list(string)
                sku_tier                      = string
                base_policy_id                = string
                private_ip_ranges             = list(string)
                threat_intelligence_mode      = string
                threat_intelligence_allowlist = list(string)
                availability_zones = object({
                  zone_1 = bool
                  zone_2 = bool
                  zone_3 = bool
                })
              })
            })

```

References:

- Azure firewall premium certs and integration with AKV: https://learn.microsoft.com/en-us/azure/firewall/premium-certificates

#### Describe the solution you'd like

- Configuration support and examples for specifying the AKV CA certificate path for Azure Firewall

#### Additional context

- For enterprise deployments, it should be possible (desirable) for Azure Firewall to be deployed and configured with the correct CA certificate for TLS encryption from inception instead of as a post-deploy operation.

**Comment by krowlandson** [comment]
Thank you for logging this @archmangler... this feature is enabled via the Azure Firewall Policy resource, and wasn't available when we added this to the module.

Given it's now available in all supported versions, I'm happy to add this. We will probably include this in the inputs as part of updating the module to support `optional()` attribute types.

**Comment by krowlandson** [comment]
@archmangler this is now available through [release v3.0.0](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/releases/tag/v3.0.0).

For now, to set this please use the `advanced` block in `configure_connectivity_settings`:

```hcl
    advanced = {
      custom_settings_by_resource_type = {
        azurerm_firewall_policy = {
          connectivity = {
            location = {                 # Replace 'location' with the location you are deploying this resource to; e.g. 'eastus'
              tls_certificate = [
                key_vault_secret_id = "" # (Required) The ID of the Key Vault, where the secret or certificate is stored.
                name                = "" # (Required) The name of the certificate.
              ]
            }
          }
        }
      }
    }
```

We may move this into the main schema at some point if demand is there.
